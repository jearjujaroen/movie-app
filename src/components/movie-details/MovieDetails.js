import React from 'react';
import PropTypes from 'prop-types';
import styles from './MovieDetails.module.css';
import { MovieCredit } from '../movie-credit/MovieCredit';

export const MovieDetails = ({
  cast = [],
  movie = {},
  movieId = null,
  baseBackdropUrl = '',
  baseProfileUrl = '',
}) => {
  return (
    <div className={styles.container}>
      <h1>{movie.title}</h1>
      <img
        className={styles.imgBackdrop}
        src={`${baseBackdropUrl}${movie.backdrop_path}`}
        alt={movie.title}
      />
      <h4>Overview:</h4>
      <div>{movie.overview}</div>
      <h4>Main Cast</h4>
      <div className={styles.movieCreditsWrapper}>
        <MovieCredit
          cast={cast}
          movieId={movieId}
          baseProfileUrl={baseProfileUrl}
        />
      </div>
    </div>
  );
};

MovieDetails.propTypes = {
  cast: PropTypes.array,
  movie: PropTypes.object.isRequired,
  movieId: PropTypes.string.isRequired,
  baseBackdropUrl: PropTypes.string.isRequired,
  baseProfileUrl: PropTypes.string.isRequired,
};
