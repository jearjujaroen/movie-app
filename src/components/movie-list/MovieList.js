import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './MovieList.module.css';

export const MovieList = ({ movies = [], basePosterUrl = '' }) => {
  //TODO: refactor this into smaller components?
  return (
    <div className={styles.container}>
      {movies && movies.length > 0 ? (
        <ol className={styles.wrapper}>
          {movies.map(movie => (
            <li className={styles.box} key={movie.id}>
              {movie.poster_path && (
                <div className={styles.cardContainer}>
                  <Link to={`/movie/${movie.id}`}>
                    <img
                      src={`${basePosterUrl}${movie.poster_path}`}
                      alt={movie.title}
                    />
                    <div className={styles.descriptionText}>
                      <p>{movie.vote_average}</p>
                    </div>
                  </Link>
                </div>
              )}
            </li>
          ))}
        </ol>
      ) : (
        <p>No results!</p>
      )}
    </div>
  );
};

MovieList.propTypes = {
  movies: PropTypes.array.isRequired,
  basePosterUrl: PropTypes.string.isRequired,
};
