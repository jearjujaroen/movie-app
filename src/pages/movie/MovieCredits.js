import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

const CastProfiles = ({ casts, baseProfileUrl }) =>
  casts
    .filter(cast => cast.profile_path !== null)
    .map(cast => (
      <img
        key={cast.cast_id}
        src={`${baseProfileUrl}${cast.profile_path}`}
        alt='cast'
      />
    ));

export const MovieCredits = ({ movieId, baseProfileUrl }) => {
  const [casts, setCasts] = useState([]);
  useEffect(() => {
    //TODO: refactor fetchMovieCredits api into APIUtils
    const fetchMovieCredits = async () => {
      const movieCreditResp = await (
        await fetch(
          `https://api.themoviedb.org/3/movie/${movieId}/credits?api_key=${process.env.REACT_APP_MOVIE_DB_KEY}`
        )
      ).json();
      const movieCast = movieCreditResp.cast;
      setCasts(movieCast);
    };

    fetchMovieCredits();
  }, [movieId]);

  return (
    <>
      <CastProfiles {...{ casts, baseProfileUrl }} />
    </>
  );
};

MovieCredits.propTypes = {
  movieId: PropTypes.string.isRequired,
  baseProfileUrl: PropTypes.string.isRequired,
};
