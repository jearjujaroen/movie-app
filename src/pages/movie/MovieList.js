import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './MovieList.module.css';

import { LazyImage } from '../../components/LazyImage/Image';

export const MovieList = ({ movies, basePosterUrl }) => {
  return (
    <div className={styles.container}>
      {movies && movies.length > 0 ? (
        <ol className={styles.wrapper}>
          {movies.map(movie => (
            <li className={styles.box} key={movie.id}>
              {movie.poster_path && (
                <div className={styles.cardContainer}>
                  <Link to={`/movie/${movie.id}`}>
                    <LazyImage src={`${basePosterUrl}${movie.poster_path}`} />
                    <div className={styles.descriptionText}>
                      <p>{movie.vote_average}</p>
                    </div>
                  </Link>
                </div>
              )}
            </li>
          ))}
        </ol>
      ) : (
        <p>No results!</p>
      )}
    </div>
  );
};

MovieList.propTypes = {
  movies: PropTypes.array.isRequired,
  basePosterUrl: PropTypes.string.isRequired,
};
