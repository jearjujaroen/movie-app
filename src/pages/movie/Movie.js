import React from 'react';
import PropTypes from 'prop-types';
import { MovieDetails } from './MovieDetails';
import { Route, useRouteMatch } from 'react-router-dom';

export const Movie = ({ baseBackdropUrl, baseProfileUrl }) => {
  let { path } = useRouteMatch();
  return (
    <>
      <Route path={`${path}/:movieId`}>
        <MovieDetails
          baseBackdropUrl={baseBackdropUrl}
          baseProfileUrl={baseProfileUrl}
        />
      </Route>
    </>
  );
};

Movie.propTypes = {
  baseBackdropUrl: PropTypes.string.isRequired,
  baseProfileUrl: PropTypes.string.isRequired,
};
