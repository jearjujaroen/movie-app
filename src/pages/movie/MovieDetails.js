import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useParams } from 'react-router-dom';
import styles from './MovieDetails.module.css';
import { MovieCredits } from './MovieCredits';

export const MovieDetails = ({ baseBackdropUrl, baseProfileUrl }) => {
  const [movie, setMovie] = useState({});
  let { movieId } = useParams();

  useEffect(() => {
    const fetchMovieDetails = async () => {
      try {
        const movieDetailsResp = await (
          await fetch(
            `https://api.themoviedb.org/3/movie/${movieId}?api_key=${process.env.REACT_APP_MOVIE_DB_KEY}&language=en-US`
          )
        ).json();
        setMovie(movieDetailsResp);
      } catch (e) {
        console.log(e);
      }
    };
    fetchMovieDetails();
  }, [movieId]);

  return (
    <div className={styles.container}>
      <h1>{movie.title}</h1>
      <img
        className={styles.imgBackdrop}
        src={`${baseBackdropUrl}${movie.backdrop_path}`}
        alt={movie.title}
      />
      <h4>Overview:</h4>
      <div>{movie.overview}</div>
      <h4>Main Cast</h4>
      <div className={styles.movieCreditsWrapper}>
        <MovieCredits movieId={movieId} baseProfileUrl={baseProfileUrl} />
      </div>
    </div>
  );
};

MovieDetails.propTypes = {
  baseBackdropUrl: PropTypes.string.isRequired,
  baseProfileUrl: PropTypes.string.isRequired,
};
