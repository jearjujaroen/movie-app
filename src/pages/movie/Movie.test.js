import React from 'react';
import renderer from 'react-test-renderer';
import { Movie } from './Movie';
import { MovieDetails } from './MovieDetails';
import { MovieList } from './MovieList';
import { MovieCredits } from './MovieCredits';
import { Route, MemoryRouter } from 'react-router-dom';

const baseProfileUrl = '';

describe('Movie', () => {
  test('snapshot renders', () => {
    const baseBackdropUrl = '';

    const component = renderer.create(
      <MemoryRouter initialEntries={['/1']}>
        <Route path={`/:movieId`}>
          <Movie {...{ baseBackdropUrl, baseProfileUrl }} />
        </Route>
      </MemoryRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('MovieDetails', () => {
  test('snapshot renders', () => {
    const baseBackdropUrl = '';

    const component = renderer.create(
      <MemoryRouter initialEntries={['/1']}>
        <Route path={`/:movieId`}>
          <MovieDetails {...{ baseBackdropUrl, baseProfileUrl }} />
        </Route>
      </MemoryRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('MovieList', () => {
  test('snapshot renders', () => {
    const basePosterUrl = '';
    const movies = [];

    const component = renderer.create(
      <MovieList {...{ basePosterUrl, movies }} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('MovieCredits', () => {
  test('snapshot renders', () => {
    const movieId = '2';

    const component = renderer.create(
      <MovieCredits {...{ baseProfileUrl, movieId }} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
