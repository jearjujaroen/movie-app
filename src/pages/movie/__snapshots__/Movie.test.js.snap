// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie snapshot renders 1`] = `null`;

exports[`MovieCredits snapshot renders 1`] = `null`;

exports[`MovieDetails snapshot renders 1`] = `
<div
  className="container"
>
  <h1 />
  <img
    className="imgBackdrop"
    src="undefined"
  />
  <h4>
    Overview:
  </h4>
  <div />
  <h4>
    Main Cast
  </h4>
  <div
    className="movieCreditsWrapper"
  />
</div>
`;

exports[`MovieList snapshot renders 1`] = `
<div
  className="container"
>
  <p>
    No results!
  </p>
</div>
`;
